FHEM/00_SIGNALduino.pm

```diff
@@ -1964,14 +1964,13 @@ sub SIGNALduino_moduleMatch
 	
 	if (!defined($modMatchRegex) || $dmsg =~ m/$modMatchRegex/) {
 		Debug "$name: modmatch passed for: $dmsg" if ($debug);
-		my $developATTR = AttrVal($name,"development","");
 		my $developID = SIGNALduino_getProtoProp($id,"developId","");
 		if ($developID eq "m") {
-			if ( length($developATTR) > 0 && $developATTR =~ m/$developID$id/) { 
-				return 1; #   return 1 da modulematch gefunden wurde			
+			if (defined($hash->{IDsNoDispatch}) && $hash->{IDsNoDispatch} =~ m/(^$id,|,$id,|,$id$)/) {	# kein dispatch wenn die Id nicht im Attribut whitelist_IDs steht
```

Comment:
`$hash `ist hier nicht definiert.

Wäre an dieser Stelle vermutlich auch besser über die API mit `InternalVal` zu arbeiten:

`if (InternalVal($name,"IDsNoDispatch",0) && InternalVal($name,"IDsNoDispatch","") =~ m/(?:^$id,|,$id,|,$id$)/) ) {  # kein dispatch wenn die Id nicht im Attribut whitelist_IDs steht
`

Leider wird die Regex bei jeder Abfrage neu compiliert.  Das könnte man lösen, indem man die ganze Schleife in ein eval packt und das abspeichert.

Reply:
so funktioniert es auch. Index ist die schnellste Möglichkeit um zu suchen 

```
my $IDsNoDispatch = "," . InternalVal($name,"IDsNoDispatch","") . ",";
if ($IDsNoDispatch ne ",," && index($IDsNoDispatch, ",$id,") >= 0) {
```

Reply:
ja, mit den Kommas vorne und hinten geht es natürlich auch:)